src/app/services/pricing.service.ts

```diff
@@ -172,16 +174,38 @@ export class PricingService {
         return pricing;
     }
 
+    /**
+     * Sets the software prices using meter IDs
+     * @param meters RateCardMeter[]
+     * @param pricing BatchPricing
+     */
     private _processSoftwaresPricings(meters: RateCardMeter[], pricing: BatchPricing) {
         for (const meter of meters) {
             if (meter.MeterId in softwareMeterId) {
                 const software = softwareMeterId[meter.MeterId];
-                const perCore = meter.MeterName.toLowerCase().includes("1 vcpu");
-                pricing.softwares.add(software, meter.MeterRates["0"], perCore);
+                this._setBillingUnit(meter);
+                pricing.softwares.add(software, meter.MeterRates["0"], this.billingUnit);
             }
         }
     }
 
+    /**
+     * Sets billing unit of software prices
+     * @param meter RateCardMeter
+     */
+    private _setBillingUnit(meter: RateCardMeter) {
+        switch (meter.MeterName.toLowerCase()) {
+            case "1 vcpu vm license":
```

Comment:
Do we know how reliable these meter names are?

Reply:
We should probably be using the `meter.Unit`  property to infer this? Looks like their entire API is strings though rather than enums so they could change pretty much whatever they want

Reply:
The `meter.Unit` property only printed out "1 hour" when I tested it. I will keep it as `meter.MeterName` for now unless there are other suggestions.